datalake/backend/tests/test_dataset_versions_endpoint.py

```diff
@@ -0,0 +1,104 @@
+"""
+Dataset Versions endpoint Lambda function tests. Working Data Lake AWS environment is
+required (run '$ cdk deploy' before running tests).
+"""
+
+import logging
+
+import _pytest
+from pytest import mark
+
+from ..endpoints.dataset_versions import entrypoint
+from .utils import Dataset, any_dataset_id, any_lambda_context, any_s3_url, any_valid_dataset_type
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def test_should_fail_if_request_not_containing_method() -> None:
```

Comment:
"should fail" is a bad naming convention. Basically, there are an infinite number of ways a piece of code could be said to "fail," and we want the test names to be specific enough to understand which failure mode is under test. In this case, what we are actually verifying is that a missing method results in a "bad request" response, so we could use a name like "test_should_return_required_property_error_when_missing_http_method`.